nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-common/src/main/java/org/apache/nifi/processors/iceberg/AbstractIcebergProcessor.java

```diff
@@ -68,35 +70,34 @@ public abstract class AbstractIcebergProcessor extends AbstractProcessor impleme
             .description("A FlowFile is routed to this relationship if the operation failed and retrying the operation will also fail, such as an invalid data or schema.")
             .build();
 
-    protected final AtomicReference<KerberosUser> kerberosUserReference = new AtomicReference<>();
+    protected static final AtomicReference<KerberosUser> kerberosUserReference = new AtomicReference<>();
     protected final AtomicReference<UserGroupInformation> ugiReference = new AtomicReference<>();
 
     @OnScheduled
     public void onScheduled(final ProcessContext context) {
         initKerberosCredentials(context);
     }
 
-    protected void initKerberosCredentials(ProcessContext context) {
+    protected synchronized void initKerberosCredentials(ProcessContext context) {
         final KerberosUserService kerberosUserService = context.getProperty(KERBEROS_USER_SERVICE).asControllerService(KerberosUserService.class);
         final IcebergCatalogService catalogService = context.getProperty(CATALOG).asControllerService(IcebergCatalogService.class);
 
         if (kerberosUserService != null) {
-            final KerberosUser kerberosUser = kerberosUserService.createKerberosUser();
-            kerberosUserReference.set(kerberosUser);
+            KerberosUser kerberosUser;
+            if (kerberosUserReference.get() == null) {
+                kerberosUser = kerberosUserService.createKerberosUser();
+            } else {
+                kerberosUser = kerberosUserReference.get();
+            }
```

Comment:
Just call `get()` once
```suggestion
            KerberosUser kerberosUser = kerberosUserReference.get() ;
            if (kerberosUser == null) {
                kerberosUser = kerberosUserService.createKerberosUser();
            } 
```